pyaerocom/aeroval/obsentry.py

```diff
@@ -101,22 +197,20 @@ def check_add_obs(self):
                 )
             if self.obs_id not in const.OBS_IDS_UNGRIDDED:
                 try:
-                    const.add_ungridded_post_dataset(**self)
+                    const.add_ungridded_post_dataset(**self.model_dump())
                 except Exception:
                     raise InitialisationError(
                         f"Cannot initialise auxiliary reading setup for {self.obs_id}. "
                         f"Reason:\n{format_exc()}"
                     )
 
-    def get_all_vars(self) -> list:
+    def get_all_vars(self) -> tuple[str, ...]:
```

Comment:
Part of the validator of `obs_vars` is that if you provide `ObsEntry` with a string, it get coerced into a tuple with one element, so this method will only return a tuple containing (a) string(s)

Reply:
Should obs_vars be typehinted as just tuple then?

Reply:
No, because that is the type hint of what can be provided, not it's internal representation / what gets returned. If we type hint it as just a tuple, users have to always provide a tuple of strings, which would break the interface.

Reply:
That's fair but this breaks a lot of the tooling around types such as autocomplete since vscode doesn't know what the real type of the attributes are.

Maybe we can look into doing something like this at a later point: https://github.com/pydantic/pydantic/discussions/4725#discussioncomment-4337576